contrib/opencensus-ext-azure/opencensus/ext/azure/log_exporter/__init__.py

```diff
@@ -106,6 +106,20 @@ def log_record_to_envelope(self, record):
         raise NotImplementedError  # pragma: NO COVER
 
     def flush(self, timeout=None):
+        if self._queue.is_empty():
+            return
+
+        # We must check the worker thread is alive, because otherwise flush
+        # is useless. Also, it would deadlock if not timeout is given, and the
+        # queue isn't empty.
+        # This is a very possible scenario during process termination, when
+        # atexit first calls handler.close() and then logging.shutdown(),
+        # that in turn calls handler.flush() without arguments.
+        if not self._worker.is_alive():
+            logger.warning("Can't flush %s, worker thread is dead. ",
+                           "Pending messages will be lost.", self)
```

Comment:
probably meant to concatenate?

```suggestion
            logger.warning("Can't flush %s, worker thread is dead. "
                           "Pending messages will be lost.", self)
```

Reply:
Indeed, good catch